examples/plot_srm_alignment.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+
+"""
+Hyperalignment-base prediction using Hugo Richard's FastSRM.
+See article : https://doi.org/10.1162/imag_a_00032
+
+==========================
+
+In this tutorial, we show how to better predict new contrasts for a target
+subject using many source subjects corresponding contrasts. For this purpose,
+we create a template to which we align the target subject, using shared information.
+We then predict new images for the target and compare them to a baseline.
+
+We mostly rely on Python common packages and on nilearn to handle
+functional data in a clean fashion.
+
+
+To run this example, you must launch IPython via ``ipython
+--matplotlib`` in a terminal, or use ``jupyter-notebook``.
+
+.. contents:: **Contents**
+    :local:
+    :depth: 1
+
+"""
+
+###############################################################################
+# Retrieve the data
+# -----------------
+# In this example we use the IBC dataset, which includes a large number of
+# different contrasts maps for 12 subjects.
+# We download the images for subjects sub-01, sub-02, sub-04, sub-05, sub-06
+# and sub-07 (or retrieve them if they were already downloaded).
+# imgs is the list of paths to available statistical images for each subjects.
+# df is a dataframe with metadata about each of them.
+# mask is a binary image used to extract grey matter regions.
+#
+
+from fmralign.fetch_example_data import fetch_ibc_subjects_contrasts
+
+imgs, df, mask_img = fetch_ibc_subjects_contrasts(
+    ["sub-01", "sub-02", "sub-04", "sub-05", "sub-06", "sub-07"]
+)
+
+###############################################################################
+# Definine a masker
+# -----------------
+# We define a nilearn masker that will be used to handle relevant data.
+#   For more information, visit :
+#   'http://nilearn.github.io/manipulating_images/masker_objects.html'
+#
+
+from nilearn.maskers import NiftiMasker
+
+masker = NiftiMasker(mask_img=mask_img).fit()
+
+###############################################################################
+# Prepare the data
+# ----------------
+# For each subject, we will use two series of contrasts acquired during
+# two independent sessions with a different phase encoding:
+# Antero-posterior(AP) or Postero-anterior(PA).
+#
+
+
+# To infer a template for subjects sub-01 to sub-06 for both AP and PA data,
+# we make a list of 4D niimgs from our list of list of files containing 3D images
+
+from nilearn.image import concat_imgs
+
+template_train = []
+for i in range(6):
```

Comment:
Is there a reason why we want to have `sub-07` in `template_train` at all, rather than only in `target_train` and `target_test` ? I know we exclude the last entry later so that it's not actually included in e.g., `average_img`, but that's one extra bit of overhead for the reader to keep in mind.

Reply:
I don't understand ? We need the data from sub-07 in the training to get its basis, so we need to concat the imgs at some point. Maybe you mean changing the same of this list to simply imgs ? But it would be inconsistant with the INT experiment ?

Reply:
What would be your suggestion @emdupre  ?

Reply:
Oh, I see it would be to use `add_subjects`. @denisfouchard can you try and do that ?